File: Pot.cpp
#include <GL/glut.h>

float rotationAngle = 0.0f; //rotation angle

void display() {
    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
    glLoadIdentity(); 
    gluLookAt(0, 0, 5, 0, 0, 0, 0, 1, 0); //set viewpoint
    glRotatef(rotationAngle, 1, 1, 1); //rotate teapot 
    glutWireTeapot(1.0); //draw teapot
    glutSwapBuffers();
}

void reshape(int width, int height) { 
    glMatrixMode(GL_PROJECTION);
    gluPerspective(45, (double)width / height, 0.1, 50.0);
    glMatrixMode(GL_MODELVIEW);
}

//rotation logic
void idle() {
    rotationAngle += 0.5f;
    //set angle to 0 if > 360
    if (rotationAngle > 360.0f) {
        rotationAngle -= 360.0f;
    }
    glutPostRedisplay(); //fluid rotation
}


int main(int argc, char** argv) {
    glutInit(&argc, argv);
    glutInitDisplayMode(GLUT_DOUBLE | GLUT_RGB | GLUT_DEPTH);
    glutCreateWindow("Utah Teapot");
    glEnable(GL_DEPTH_TEST);
    glutDisplayFunc(display);
    glutReshapeFunc(reshape);
    glutIdleFunc(idle);
    glutMainLoop();
    return 0;
}

File: Pot.py
from OpenGL.GL import *
from OpenGL.GLUT import *
import pygame
from pygame.locals import *
from OpenGL.GLU import gluPerspective
glutInit()

#initialize the display
def initialize():
    glClearColor(0, 0, 0, 1)
    gluPerspective(45, (display_size[0] / display_size[1]), 0.1, 50.0)
    glTranslatef(0.0, 0.0, -5)

def display():
    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT)

    glutWireTeapot(1.0) #draw the teapot
    pygame.display.flip()

def main():
    pygame.init()
    global display_size
    display_size = (800, 600)
    pygame.display.set_mode(display_size, DOUBLEBUF | OPENGL)
    pygame.display.set_caption("Utah Teapot")

    glEnable(GL_DEPTH_TEST)
    initialize()

    while True:
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                pygame.quit()
                quit()

        glRotatef(1, 3, 1, 1)
        display()
        pygame.time.wait(10)

if __name__ == "__main__":
    main()

